### DISCUSSION TITLE
Cannot set global atom outside provider tree?

### DISCUSSION THREAD

**Discussion by devnomic** [pmndrs/jotai#1771] [General] [answered]
I have a generic component that hold it's own atom, and in that component the atom is initialized using provider store.

However inside that component i also need to modify global state atom outside the provider.
Why it doesn't work and how to fix this?

Reproduction:
https://codesandbox.io/s/hopeful-shirley-8lw6dh?file=/src/App.tsx

**Comment by dai-shi** [comment, marked as answer]
You may have a misconception about global atom.
Atoms are just definitions and atom values are stored in a store.
So, you want to define a global store explicitly and specify it.
See how it works: https://codesandbox.io/s/crazy-buck-ryx209?file=/src/App.tsx

**Reply by devnomic** [reply]
Thank you!!

I didn't know we can put `store` as option parameters when using `useSetAtom`  or other atom hooks. 
The docs didn't mention it explicitly.

**Reply by dai-shi** [reply]
Yeah... some docs are still missing for v2 API and best practice.
